File: src/main/java/ru/vmsystems/template/domain/service/ProcessTypeService.java
package ru.vmsystems.template.domain.service;

import org.dozer.DozerBeanMapper;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.vmsystems.template.domain.model.ProcessTypeEntity;
import ru.vmsystems.template.infrastructure.persistence.ProcessTypeRepository;
import ru.vmsystems.template.interfaces.dto.ProcessTypeDto;

import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@Service
public class ProcessTypeService extends BackService {

    @Autowired
    private ProcessTypeRepository repository;
    @Autowired
    private DozerBeanMapper mapper;

    public List<ProcessTypeDto> get() {
        return repository.getByCompanyId(getCompanyId()).stream()
                .map(entity -> mapper.map(entity, ProcessTypeDto.class))
                .collect(Collectors.toList());
    }

    public Optional<ProcessTypeDto> get(@NotNull Long id) {
        ProcessTypeEntity entity = repository.findOne(id);
        if (entity == null) {
            return Optional.empty();
        }

        return Optional.of(mapper.map(entity, ProcessTypeDto.class));
    }

    @NotNull
    public ProcessTypeDto create(@NotNull ProcessTypeDto dto) {
        return update(null, dto);
    }

    @NotNull
    public ProcessTypeDto update(@Nullable Long id, @NotNull ProcessTypeDto dto) {

        dto.setId(id);
        ProcessTypeEntity entity = mapper.map(dto, ProcessTypeEntity.class);
        entity.setCompany(getCompany());
        entity = repository.save(entity);

        return mapper.map(entity, ProcessTypeDto.class);
    }

    public void delete(Long orderId) {
        repository.delete(orderId);
    }
}

File: src/main/java/ru/vmsystems/glasscloud/domain/process/type/ProcessTypeService.kt
package ru.vmsystems.glasscloud.domain.process.type

import org.springframework.stereotype.Service
import ru.vmsystems.glasscloud.domain.BackService
import ru.vmsystems.glasscloud.domain.SessionService
import ru.vmsystems.glasscloud.domain.process.ProcessTypeDto
import ru.vmsystems.glasscloud.domain.process.ProcessTypeEntity
import ru.vmsystems.glasscloud.domain.process.ProcessTypeRepository
import ru.vmsystems.glasscloud.handler.BusinessException
import java.util.*

@Service
class ProcessTypeService(
        private val repository: ProcessTypeRepository,
        private val sessionService: SessionService
) : BackService() {

    fun getAll(): List<ProcessTypeDto> {
        return repository.getByCompanyId(sessionService.currentCompanyId)
                .map { it.transform() }
                .toList()
    }

    fun get(id: UUID): ProcessTypeDto {
        val entity = repository.getById(id) ?: throw BusinessException("Не найден тип работ")
        return entity.transform()
    }

    fun save(dto: ProcessTypeDto) {
        val entity = dto.transform(sessionService.currentCompanyId)
        repository.save(entity)
    }

    fun delete(id: UUID) {
        repository.deleteById(id)
    }
}

fun ProcessTypeDto.transform(currentCompanyId: UUID? = null): ProcessTypeEntity {
    return ProcessTypeEntity(
            id = id,
            name = name,
            deleted = deleted,
            description = description,
            companyId = currentCompanyId ?: companyId
    )
}

fun ProcessTypeEntity.transform(): ProcessTypeDto {
    return ProcessTypeDto(
            id = id!!,
            name = name,
            deleted = deleted,
            description = description,
            companyId = companyId
    )
}
